lib/postgresql.js

```diff
@@ -354,34 +355,48 @@ PostgreSQL.prototype.applyPagination = function(model, stmt, filter) {
   return stmt.merge(limitClause);
 };
 
-PostgreSQL.prototype.buildExpression =
-  function(columnName, operator, columnValue, propertyValue) {
-    if (propertyValue instanceof RegExp) {
-      columnValue = "'" + propertyValue.source + "'";
-      if (propertyValue.ignoreCase) {
-        return new ParameterizedSQL(columnName + ' ~* ?', [columnValue]);
-      } else {
-        return new ParameterizedSQL(columnName + ' ~ ?', [columnValue]);
-      }
+PostgreSQL.prototype.buildExpression = function(columnName, operator,
+    columnValue, propertyValue) {
+  if (operator === 'regexp') {
+    var regex = utils.getRegExpExpression(utils.toRegExp(columnValue));
```

Comment:
The regexp value is already an instance of `RegExp`. Why further processing here?